specviz/widgets/plotting.py

```diff
@@ -503,3 +534,270 @@ def _on_remove_linear_region(self):
         self.removeItem(self._selected_region)
         self._selected_region = None
         self._region_text_item.setText("")
+
+    def set_units(self, data_unit, spectral_axis_unit):
+        self._data_unit = data_unit
+        self._spectral_axis_unit = spectral_axis_unit
+
+        self._plot_item.setLabel('bottom', units=self.spectral_axis_unit)
+        self._plot_item.setLabel('left', units=self.data_unit)
+
+        self.autoRange()
+
+
+class UnitChangeDialog(QDialog):
```

Comment:
The `UnitChangeDialog` should be defined in its own file, probably in the `specviz/widgets` directory.